frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx

```diff
@@ -207,79 +50,155 @@ const DatasetRow: FC<Props> = ({
   file,
   invalidateCollectionQuery,
 }) => {
-  const {
-    tissue,
-    assay,
-    disease,
-    organism,
-    cell_count,
-  } = aggregateDatasetsMetadata([dataset]);
   const queryCache = useQueryCache();
+
   const queryResult = useDatasetStatus(
     dataset.id,
     checkIfLoading(dataset.processing_status)
   );
-  const [lastUploadProgress, setLastUploadProgress] = useState(
-    INITIAL_UPLOAD_PROGRESS
-  );
 
-  if (queryResult.isError) console.error(queryResult.error);
+  const datasetStatus = queryResult.data || dataset.processing_status;
 
-  const datasetStatus = queryResult.data ?? dataset.processing_status;
-  const isLoading = checkIfLoading(datasetStatus);
-  const hasFailed = checkIfFailed(datasetStatus);
+  const { upload_progress } = datasetStatus;
 
-  // `nameIsPopulated` is a temp-workaround for no "complete status"
-  // const isComplete = checkIfComplete(datasetStatus);
-  const nameIsPopulated = dataset.name !== undefined && dataset.name !== "";
+  const [uploadProgress, setUploadProgress] = useState(upload_progress);
```

Comment:
Set default `uploadProgress` to be `upload_progress` instead of `-1`, I think that works and saves us one `setState` cycle?